shell/browser/native_window_mac.mm

```diff
@@ -1403,15 +1403,35 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   original_frame_ = [window_ frame];
 }
 
-void NativeWindowMac::SetVibrancy(const std::string& type) {
+void NativeWindowMac::SetVibrancy(const std::string& type, bool animate) {
   NSVisualEffectView* vibrantView = [window_ vibrantView];
 
   if (type.empty()) {
     if (vibrantView == nil)
       return;
 
-    [vibrantView removeFromSuperview];
-    [window_ setVibrantView:nil];
+    vibrancy_type_ = type;
+
+    if (animate) {
+      __weak auto weak_delegate = window_delegate_.get();
+      [NSAnimationContext
+          runAnimationGroup:^(NSAnimationContext* context) {
+            context.duration = 0.3;
```

Comment:
A hard coded value might not be enough for all apps, some apps might want a slow fading in animation for startup effect, and some might want a short animation for things like notifications.

Reply:
added an `animationDuration` prop to the options object on the API